консольное приложение 2 спринт

OlegQt opened the pull request:
Реализовал через массив, мне показалось это логичным.

Commit:
Remove classes

Commit:
Change project name

OlegQt committed:
Функция ввода товаров и проверки корректности ввода готова

OlegQt committed:
Добавил функцию GetWordEnd, которая анализирует единицы входящию в цифру и возврщает слово рубль в нужном падеже.

OlegQt committed:
Проверка на отрицательную цену

OlegQt committed:
Вроде все работает

OlegQt committed:
Проверка на отрицательное число человек

gusar-off commented on src/main/java/Calc.java:
```diff
@@ -0,0 +1,83 @@
+            }
+            System.out.println("Товар успешно добавлен, хотите продолжить? \n для завершения введите \"завершить\" ");
+            str = pScanner.next();
+            if(str.equalsIgnoreCase("завершить")) break;// Выходим, если введено завершить
```
"завершить" лучше вынести в статическую константу

gusar-off commented on src/main/java/Calc.java:
```diff
@@ -0,0 +1,83 @@
+    ArrayList<Commodity> pList = new ArrayList<Commodity>();
+    double sum = 0; // Полная сумма всех товаров
+    public void startCalc()
+    {
```
В кодстиле джава рекомендуется не переносить скобочку на следующую строчку
```
public void startCalc() {
```
и так лучше сделать везде.

gusar-off commented on src/main/java/Calc.java:
```diff
@@ -0,0 +1,83 @@
+    {
+        System.out.println("Добавленные товары:");
+        this.sum = 0; // Перед расчетом обнулим сумму товаров на всякий случай
+        for (int i = 0; i < pList.size(); i++) {
```
Тут лучше заменить на цикл foreach
```
for (Commodity item : pList) {
```

OlegQt replied:
Не знаю, можно ли здесь вопрос писать. Как этот цикл внутри выполняется: он так же прогоняет поштучно и по времени выходит то же самое, или кроме короткого синтаксиса в foreach есть выигрыш по скорости чтения?

gusar-off commented on src/main/java/Calc.java:
```diff
@@ -0,0 +1,83 @@
+    public void calculate(int personQuantity)
+    {
+        double money = this.sum/personQuantity;
+        //double money = 5.9589;
```
Закомментированный код лучше убрать.

gusar-off commented on src/main/java/Calc.java:
```diff
@@ -0,0 +1,83 @@
+    {
+        String strOst="";
+        num = Math.floor(num);
+        int ost =  (int)(num % 10);
```
Немного некорректно рассчитывается окончание. Получается если цена будет 114 рублей, метод вернет "рубля". Правильный алгоритм звучит так:
Проверить лежит ли остаток от деления на 100 в интервале от 11 до 14 включительно, если да, то возвращаем "рублей", если нет, то берем остаток от деления на 10 и прогоняем с твоими проверками.
Тем самым учтем, что числа заканчивающиеся на 11-12-13-14 и заканчивающиеся на 1-2-3-4, будут иметь разные окончания.

gusar-off commented on src/main/java/Calc.java:
```diff
@@ -0,0 +1,83 @@
+                strOst="рублей";
+                break;
+        }
+        if(num>9&&num<21) strOst="рублей";
```
и это тогда лишнее

OlegQt committed:
Исправления:
Убрал лишние закомментированные строки
Переделал цикл просмотра листа товаров
Поправил функцию проверки итоговой суммы и ыбора падежа слова "рубль"

OlegQt committed:
Исправления:
Убрал лишние закомментированные строки
Переделал цикл просмотра листа товаров
Поправил функцию проверки итоговой суммы и ыбора падежа слова "рубль"

gusar-off commented on src/main/java/Calc.java:
```diff
@@ -0,0 +1,79 @@
+                return "рублей";
+        }
+    }
+    public void releaseArray(){
```
Метод не используется, лучше тогда убрать

gusar-off commented on src/main/java/Calc.java:
```diff
@@ -0,0 +1,79 @@
+        System.out.println("Добавленные товары:");
+        this.sum = 0; // Перед расчетом обнулим сумму товаров на всякий случай
+        for ( Commodity i:pList){
+            i.showCommodity();
```
Лучше давать осмысленные названия переменным, например тут можно назвать `item`
